pyproject.toml

```diff
@@ -114,7 +126,7 @@ testpaths = ["./qiskit_addon_sqd/", "./test/"]
 [tool.ruff]
 line-length = 100
 src = ["qiskit_addon_sqd", "test"]
-target-version = "py39"
+target-version = "py310"
```

Comment:
Looks like this causes one file to be reformatted. Feel free to revert this or let me know if I should reformat that file as part of this PR.

Reply:
No, I think we need to upgrade this since one of our dependencies (jax) doesnt work with py39.